### DISCUSSION TITLE
Handling Beta Updates

### DISCUSSION THREAD

**Discussion by davidnorris78** [sparkle-project/Sparkle#2197] [General]
I want to introduce beta channel for updates.  I want the app to have two menu items for checking updates.  one for the default channel and another menu item for Checking for Beta updates on the beta channel only.

Is there anyway to achieve this without having to introduce another appcast.xml file.

Any help you can offer would be great.

**Comment by zorgiepoo** [comment]
That's an odd way of providing beta updates. The standard way for an app to provide them is to have a checkbox or switch in their settings somewhere to opt into beta updates, not a separate menu item (this will make things unnecessarily harder).

Sparkle also checks for updates in the background automatically without user-initiation. If you were to go with your approach, how would it know which channel to use in that case?

Nothing about this problem is really any easier regarding channels vs feeds.
